tests/unit/test_create_notebook.py

```diff
@@ -0,0 +1,297 @@
+        )
+
+        # Assertions
+        mock_logger.error.assert_called_with(
+            f"Failed to create Notebook with error code {error_code}."
+            " Check the debug logs for more details."
+        )
+        mock_logger.debug.assert_called_with(
+            f"Failed to create Notebook {notebook_name} with error broken"
+        )
+        mock_logger.info.assert_not_called()
+
+
+def test_create_notebook_failure_time_out(
+    mock_logger: MagicMock, mock_wait_for_deployment_ready: MagicMock
+) -> None:
+    """
+    Test case to verify behavior when a TimeoutError is raised.
+    """
+    notebook_name = "test-notebook"
+    notebook_image = "test-image"
+
+    # Mock the behavior of Client
+    mock_client_instance = MagicMock()
+
+    # Mock the behavior of wait_for_deployment_ready
+    mock_wait_for_deployment_ready.side_effect = TimeoutError("test-excpetion-message")
+
+    with patch("dss.create_notebook.does_dss_pvc_exist", return_value=True), patch(
+        "dss.create_notebook.does_notebook_exist", return_value=False
+    ):
+        # Call the function to test
+        create_notebook(
+            name=notebook_name, image=notebook_image, lightkube_client=mock_client_instance
+        )
+
+        # Assertions
+        mock_logger.warn.assert_called_with(
+            f"Timed out while trying to create Notebook {notebook_name}.\n"
+            "Some resources might be left in the cluster. Check the status with `dss list`."
+        )
+        mock_logger.info.assert_not_called()
+
+
+def test_create_notebook_failure_image_pull(
+    mock_logger: MagicMock, mock_wait_for_deployment_ready: MagicMock
+) -> None:
+    """
+    Test case to verify behavior when an ImagePullBackOffError is raised.
+    """
+    notebook_name = "test-notebook"
+    notebook_image = "test-image"
+
+    # Mock the behavior of Client
+    mock_client_instance = MagicMock()
+
+    # Mock the behavior of wait_for_deployment_ready
+    mock_wait_for_deployment_ready.side_effect = ImagePullBackOffError("test-excpetion-message")
+
+    with patch("dss.create_notebook.does_dss_pvc_exist", return_value=True), patch(
+        "dss.create_notebook.does_notebook_exist", return_value=False
+    ), patch("dss.create_notebook._get_notebook_image_name", return_value=notebook_image):
+        # Call the function to test
+        create_notebook(
+            name=notebook_name, image=notebook_image, lightkube_client=mock_client_instance
+        )
+
+        # Assertions
+        mock_logger.error.assert_called_with(
+            f"Timed out while trying to create Notebook {notebook_name}.\n"
+            f"Image {notebook_image} does not exist or is not accessible.\n"
+        )
+        mock_logger.info.assert_called_with(
+            "Note: You might want to use some of these recommended images:\n"
+            f"{RECOMMENDED_IMAGES_MESSAGE}"
+        )
+
+
+# def test_get_notebook_config() -> None:
```

Comment:
Should we remove this ?

Reply:
Brought it back, I commented it out by accident.